File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        final String messagePredecessor = ":";
        
        int messageStart = logLine.indexOf(messagePredecessor) + 1;
        
        return logLine.substring(messageStart).trim();
    }

    public static String logLevel(String logLine) {
        final String logLevelPredecessor = "[";
        final String logLevelSucessor = "]";

        int logLevelStart = logLine.indexOf(logLevelPredecessor) + 1;
        int logLevelEnd = logLine.indexOf(logLevelSucessor);

        return logLine.substring(logLevelStart, logLevelEnd).toLowerCase();
    }

    public static String reformat(String logLine) {
        final String newFormat = "%s (%s)";

        String message = LogLevels.message(logLine);
        String logLevel = LogLevels.logLevel(logLine);
        
        return String.format(newFormat, message, logLevel);
    }
    
}

File: csharp/log-levels/LogLevels.cs
using System;
using System.Linq;

static class LogLine
{

    private const String LogLevelPredecessor = "[";
    private const String LogLevelSuccessor = "]";
    private const String MessagePredecessor = ": ";

    public static string Message(string logLine)
    {

        const int messageIndex = 1;

        return logLine
            .Split(LogLine.MessagePredecessor)
            .ElementAt( messageIndex )
            .Trim();
    }

    public static string LogLevel(string logLine)
    {
        logLine = logLine.Trim();
        int logLevelStartIndex = 1 + logLine.IndexOf(
            LogLine.LogLevelPredecessor
        );

        int logLevelEndIndex = -1 + logLine.IndexOf(
            LogLine.LogLevelSuccessor
        );

        int logLevelLength = logLevelEndIndex - logLevelStartIndex + 1;

        return logLine
            .Substring(logLevelStartIndex, logLevelLength)
            .ToLower();

    }

    public static string Reformat(string logLine)
    {
        string logLevel = LogLine.LogLevel(logLine);
        string message = LogLine.Message(logLine);

        return $"{message} ({logLevel})";

    }
}
